### DISCUSSION TITLE
Opentelemetry using golang - When to set the exception.escaped Attribute?

### DISCUSSION THREAD

**Discussion by richardfuchs** [open-telemetry/opentelemetry-go#3555] [Q&A] [answered]
Here is what we found [online](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/exceptions/#attributes):

exception.escaped:

> SHOULD be set to true if the exception event is recorded at a point where it is known that the exception is escaping the scope of the span.

> An exception is considered to have escaped (or left) the scope of a span, if that span is ended while the exception is still logically "in flight". This may be actually "in flight" in some languages (e.g. if the exception is passed to a Context manager's __exit__ method in Python) but will usually be caught at the point of recording the exception in most languages.
> 
> It is usually not possible to determine at the point where an exception is thrown whether it will escape the scope of a span. However, it is trivial to know that an exception will escape, if one checks for an active exception just before ending the span, as done in the example above.
> 
> It follows that an exception may still escape the scope of the span even if the exception.escaped attribute was not set or set to false, since the event might have been recorded at a time where it was not clear whether the exception will escape. […]
> 

**My question**: When to set in go the exception.escaped Attribute?

Let's consider [this](https://github.com/open-telemetry/opentelemetry-go/blob/4014204d42d55084fd90b6cb859dcde14f00231f/example/fib/app.go#L62) example: 

```go
func (a *App) Poll(ctx context.Context) (uint, error) {
	_, span := otel.Tracer(name).Start(ctx, "Poll")
	defer span.End()

	a.l.Print("What Fibonacci number would you like to know: ")

	var n uint
	_, err := fmt.Fscanf(a.r, "%d\n", &n)
	if err != nil {
		span.RecordError(err)
		span.SetStatus(codes.Error, err.Error())
		return 0, err
	}

	// Store n as a string to not overflow an int64.
	nStr := strconv.FormatUint(uint64(n), 10)
	span.SetAttributes(attribute.String("request.n", nStr))

	return n, nil
}
```

1. Should we set the Attribute exception.escaped after the _span.SetStatus(codes.Error, err.Error())_ call?
2. Is a "_return 0, err_" call seen as an escaped exception?

**Comment by MrAlias** [comment, marked as answer]
The linked code shows an error, it would not need to include exception semantics.

[Here](https://github.com/open-telemetry/opentelemetry-go/blob/a724cf884287e04785eaa91513d26a6ef9699288/sdk/trace/span.go#L381-L398) is an example of an exception event that is captured on behalf of the user if a span ends with a panic.
